apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClientImpl.SingleChannel.cs

```diff
@@ -392,13 +401,34 @@ private void HandlePullMessageData(Task<bool> moveNextTask)
                 receivedMessages.Clear();
                 // Get all ack-ids, used to extend leases as required.
                 var msgIds = new HashSet<string>(msgs.Select(x => x.AckId));
+                _receiptModAckForExactlyOnceDelivery = _exactlyOnceDeliveryEnabled;
+                if (_exactlyOnceDeliveryEnabled)
```

Comment:
I think it would be simpler to just have one _exactlyOnceDeliveryEnabled check - just put the HandleExtendLease line in the "else" block below as well. That way when reading, it's basically "do all of this for the exactly once flow" or "do all of this other block for the non-exactly-once flow".